web/src/recent_view_ui.ts

```diff
@@ -794,6 +794,20 @@ export function update_topics_of_deleted_message_ids(message_ids: number[]): voi
         msgs_to_process.push(...msgs);
     }
 
+    for (const msg_id of message_ids) {
+        const msg = message_store.get(msg_id);
+        if (msg === undefined) {
+            continue;
+        }
+
+        if (msg.type === "private") {
+            const key = recent_view_util.get_key_from_message(msg);
+            recent_view_data.conversations.delete(key);
+            const msgs_in_dm = message_util.get_messages_in_dm(msg.to_user_ids);
+            msgs_to_process.push(...msgs_in_dm);
+        }
+    }
```

Comment:
This looks like we're potentially doing quadratic work; looping through the O(10K) messages in `all_messages_data` for each deleted message.

I think you just want to store up a list of conversations to process in the first loop, and then have a single loop along the lines of `get_messages_in_dm` that goes through `all_messages_data` and computes the set of messages for each DM conversation using `set[to_user_ids]` type structures.

Reply:
Makes sense! Updated!